### PR TITLE
Add react testing library setup

### PR DESCRIPTION
**Pull request by robinandeer**
Update Jest 24 => 25.
Improve a11y support in address autocomplete modal.
Add tests for address autocomplete modal.

EDIT: Jest 25 includes an upgrade to JSDOM that includes `MutationObserver` which is used in the `waitFor` test util. (Ref: https://github.com/testing-library/dom-testing-library/issues/477)

### CODE REVIEW

**Review by styrelius — COMMENTED** [review]
Really good! One thought though - wouldn't we gain more... confidence (or whatever Kent C. Dodds usually says 😄 ) if we tested the entire `AddressAutocomplete` component rather than just the `Modal` part? 🤔

File: package.json [hunk 1]
```diff
@@ -104,7 +107,7 @@
     "babel-loader": "^8.0.6",
     "concurrently": "^5.0.0",
     "graphql-tag": "^2.10.1",
-    "jest": "^24.9.0",
+    "jest": "^25.5.4",
```
[the remark this answers is not in the document]
Comment (robinandeer) [reply package.json:110]: Jest 25 includes an upgrade to JSDOM that includes MutationObserver which is used in the waitFor test util.

I believe we should upgrade ts-jest as well. Should we do that in the same PR?

File: src/Components/Actions/AddressAutocompleteAction/Modal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+import * as React from 'react'
+import { render, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import Modal from './Modal'
+
+describe('address autocomplete modal', () => {
+  test('modal toggles display of children', async () => {
+    // Arrange
```
[the remark this answers is not in the document]
Comment (robinandeer) [reply src/Components/Actions/AddressAutocompleteAction/Modal.test.tsx:13]: Sure, we could remove them. I kind of like the grouping it gives but I'm not married to it!

**Comment by robinandeer** [comment]
> Really good! One thought though - wouldn't we gain more... confidence (or whatever Kent C. Dodds usually says 😄 ) if we tested the entire `AddressAutocomplete` component rather than just the `Modal` part? 🤔

Thanks! I see this as a start only. Wanted to keep the PR small so I started with the component that I felt was best defined and self-contained. Next step would be to tackle the autocomplete component. But then we don't need to test how the modal interactions work ;)

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (robinandeer) [reply package.json:111]: It seems like it was a painless upgrade -- thanks for pointing it out! Now all jest dependencies are upgraded

[the remark this answers is not in the document]
Comment (styrelius) [reply package.json:111]: Awesome!

**Review by styrelius — APPROVED** [review]
🧑‍🔬

**Review by gustaveen — APPROVED** [review]
Fantastic that you took the time to add this 🙌 Let's start with this!

**Pull request merged**